upgrades/4.1_5.0/tams-4.1-5.0-upgrade-template.yaml

```diff
@@ -0,0 +1,212 @@
+                items.extend(scan["Items"])
+            webhooks_dict = {}
+            for item in items:
+                event = item.pop("event")
+                if item["url"] in webhooks_dict:
+                    webhooks_dict[item["url"]]["events"].append(event)
+                else:
+                    webhooks_dict[item["url"]] = {**item, "events": [event]}
+            for webhook_dict in webhooks_dict.values():
+                webhook_dict["id"] = str(uuid.uuid4())
+                webhook_dict["status"] = "created"
+                merge_webhook(webhook_dict, None)
+
+
+        def storage_etl(context, last_evaluated_key=None):
+            args = {}
+            if last_evaluated_key:
+                args["ExclusiveStartKey"] = last_evaluated_key
+            with new_storage_table.batch_writer() as batch:
+                scan = old_storage_table.scan(**args)
+                for item in scan["Items"]:
+                    batch.put_item(Item=transform_storage_item(item))
+                while (
+                    "LastEvaluatedKey" in scan
+                    and context.get_remaining_time_in_millis() > constants.LAMBDA_TIME_REMAINING
+                ):
+                    args["ExclusiveStartKey"] = scan["LastEvaluatedKey"]
+                    scan = old_storage_table.scan(**args)
+                    for item in scan["Items"]:
+                        batch.put_item(Item=transform_storage_item(item))
+                if "LastEvaluatedKey" in scan:
+                    print(
+                        "Lambda timed out before all items were processed, restart the ETL using the following last_evaluated_key value..."
+                    )
+                    print("last_evaluated_key: ", scan["LastEvaluatedKey"])
+
+
+        def tags_etl(context):
+            query = "MATCH (n:tags) WITH n, [prop IN keys(n) WHERE NOT (n[prop] STARTS WITH '\"' OR n[prop] STARTS WITH '[')] AS valid_props WHERE size(valid_props) > 0 RETURN n.`~id` AS node_id, valid_props"
+            results = execute_open_cypher_query(query)
+            for result in results["results"]:
+                if context.get_remaining_time_in_millis() <= constants.LAMBDA_TIME_REMAINING:
+                    print(
+                        "Lambda timed out before all items were processed, re-run the tags_etl to continue"
+                    )
+                    return
+                set_clauses = ", ".join(
+                    [f"n.`{prop}` = '\"' + n.`{prop}` + '\"'" for prop in result["valid_props"]]
+                )
+                set_query = f'MATCH (n:tags {{`~id`: "{result["node_id"]}"}}) SET {set_clauses} RETURN n'
+                execute_open_cypher_query(set_query)
+
+
+        # pylint: disable=unused-argument
+        def lambda_handler(event, context):
+            # Uncomment only one of following lines at a time so that each ETL process runs in a single execution isolated on it's own
```

Comment:
Personally I'm not a fan of the "uncomment, save, execute" approach. There's multiple steps and it feels a bit easier to make a mistake.

What would you think about making it a parameter in the Lambda event payload instead?